lib/modules/manager/npm/post-update/npm.spec.ts

```diff
@@ -162,6 +160,49 @@ describe('modules/manager/npm/post-update/npm', () => {
     expect(execSnapshots).toEqual([]);
   });
 
+  it('deduplicates dependencies on installation with NPM >= 7', async () => {
+    const execSnapshots = mockExecAll();
+    // package.json
+    fs.readLocalFile.mockResolvedValueOnce('{}');
+    fs.readLocalFile.mockResolvedValueOnce('package-lock-contents');
+    const postUpdateOptions = ['npmDedupe'];
+    const updates = [
+      { packageName: 'some-dep', newVersion: '1.0.1', isLockfileUpdate: false },
+    ];
+    const res = await npmHelper.generateLockFile(
+      'some-dir',
+      {},
+      'package-lock.json',
+      { postUpdateOptions },
+      updates
+    );
+    expect(fs.readLocalFile).toHaveBeenCalledTimes(2);
+    expect(res.error).toBeFalse();
+    expect(res.lockFile).toBe('package-lock-contents');
+    expect(execSnapshots).toMatchSnapshot();
```

Comment:
Instead of this, write a deep expect statement here confirming how many commands were run, plus looking for prefer dedupe flag

Reply:
Do I understand correctly that you prefer to not use snapshot test but instead assert against the content of `execSnapshot`? Something like this:

```js
expect(execSnapshots).toHaveLength(1);
expect(execSnapshots[0].cmd).toContain('--prefer-dedupe');
```

Should I refactor the `deduplicates dependencies after installation with NPM <= 6` as well? Or should we keep it snapshot-based to not touch tests for existing functionality to much in this PR?

Reply:
Ideally leave existing tests alone to make it easier to tell that they haven't been broken. But for the new test you understand correctly what I was suggesting

Reply:
I refactored the two new tests to not use snapshots. Please let me know if I should refactor the existing test `generates lock files` as well. The snapshot of that test is changing anyways significantly as the test was using `npmDedupe` post update option previously.